src/engine/GroupBy.cpp

```diff
@@ -533,7 +533,8 @@ Result::Generator GroupBy::computeResultLazily(
       // Reuse buffer if not moved out
       resultTable = std::move(outputPair.idTable_);
       resultTable.clear();
-      currentLocalVocab = LocalVocab{};
+      // Keep last local vocab for next commit.
+      currentLocalVocab = std::move(storedLocalVocabs.back());
```

Comment:
I don't think this is necessarily correct. 
If your group is very long (e.g. from 5 IdTables ago) , then you also have to store the corresponding local vocab from exactly that group.
So you have to probably have  alocal vocab that is associated with the `currentGroupBlock` and updated exactly when you do the `commitRow` call (because then you implicitly updated the next group block ,isn't that true?